src/order/spam-protection/google-recaptcha.ts

```diff
@@ -47,34 +51,52 @@ export default class GoogleRecaptcha {
     }
 
     execute(): Observable<RecaptchaResult> {
-        if (!this._event$ || !this._recaptcha) {
+        const event$ = this._event$;
+        const recaptcha = this._recaptcha;
+
+        if (!event$ || !recaptcha) {
             throw new NotInitializedError(NotInitializedErrorType.SpamProtectionNotInitialized);
         }
 
-        this._watchRecaptchaChallengeWindow(this._event$);
+        const timeout = window.setTimeout(() => {
+            window.clearInterval(interval);
 
-        this._recaptcha.execute();
+            event$.next({
+                error: new SpamProtectionNotLoadedError(),
+            });
+        }, 7000);
+
+        const interval = window.setInterval(() => {
```

Comment:
I think it'd work using `setTimeout` and `setInterval`. But you could potentially just use `retryWhen` operator to handle this scenario. i.e.:

```ts
const maxRetries = 70;
const retryInterval = 100;

return defer(() => {
    const element = document.querySelector('iframe[src*="bframe"]');

    return element ?
        of(element) :
        throwError(new SpamProtectionNotLoadedError());
})
    .pipe(
        retryWhen(errors => errors.pipe(
            delay(retryInterval),
            switchMap((error, index) =>
                index < maxRetries ? of(error) : throwError(error)
            )
        )),
        switchMap(element => {
            this._watchRecaptchaChallengeWindow(event$, element);
            recaptcha.execute();

            return event$;
        }),
        catchError(error => of({ error }))
    );
```